annileen/editor/editorgui.ixx

```diff
@@ -73,13 +73,43 @@ export namespace annileen
 			float pitch;
 			int debugScreenActive;
 		};
-		InputConfig m_InputConfig{3.0f, 2.8f, 18.0f, 0.0f, 0.0f, 0};
 
-		std::vector<const char*> m_ConsoleLoggingChannelsList;
-		std::vector<const char*> m_ConsoleLoggingLevelsList;
+		friend class ApplicationEditor;
 
+	public:
 		EditorGui() = default;
-		~EditorGui() = default;
+		~EditorGui();
+
+	private:
+		void initialize(Engine* engine);
+		void processInput(std::shared_ptr<Camera> camera, float deltaTime);
```

Comment:
i think that all functions that get a pointer argument but are only consuming that and not extending the lifetime of the pointer, should either still get a raw pointer or get a reference to the `shared_ptr`. otherwise we're just adding overhead with the refcounts for no advantage at all.